specification/dsdl/serializable_types.tex

```diff
@@ -281,51 +313,282 @@ \subsubsection{Dependencies}
     \end{minted}
 \end{remark}
 
-\subsubsection{Unions}\label{sec:dsdl_composite_tagged_unions}
+\subsubsection{Tagged unions}\label{sec:dsdl_composite_tagged_unions}
 
-Any data schema definition can be supplied with a special directive (section~\ref{sec:dsdl_directives})
+Any data type definition can be supplied with a special directive (section~\ref{sec:dsdl_directives})
 indicating that it forms a tagged union.
 
-A tagged union schema shall contain at least two field attributes.
+A tagged union type shall contain at least two field attributes.
 A tagged union shall not contain padding field attributes.
 
 The value of a tagged union object is a function of the field attribute which value it is currently holding
 and the value of the field attribute itself.
 
+To avoid ambiguity, a data type that is not a tagged union is referred to as a \emph{structure}.
+
+\subsubsection{Alignment and cumulative bit length set}\label{sec:dsdl_composite_alignment_cumulative_bls}
+
+The alignment of composite types is one byte (8 bits)\footnote{%
+    Regardless of the content.
+    It follows that empty composites can be inserted arbitrarily to force byte alignment
+    of the next field(s) at runtime.
+}.
+
+Per the definitions given in \ref{sec:dsdl_serializable_alignment_padding},
+a serialized representation of a composite type is padded to 8 bits by inserting padding bits
+after its last element until the resulting length is a multiple of 8 bits.
+
+Given a set of field attributes, their \emph{cumulative bit length set} is computed
+by evaluating every permutation of their respective bit length sets plus the required padding.
```

Comment:
Asymptotic complexity analysis assumes that the argument approaches the infinity, but in our case `n` is guaranteed to be small because data types contain few fields. I think the current performance issues in PyDSDL are rather caused by the implementation rather than specification. Even if it is not the case, I think it is the kind of problem that is safe to shelve until later.